samples/ambient-argo/README.md

```diff
@@ -0,0 +1,99 @@
+
+|                     | Control Plane              | Data Plane        |
+| ------------------- |:--------------------------:| :----------------:|
+| **Cluster-Scoped**  | CRDs + validation          | CNI<br>ztunnel    |
+| **Workload-Scoped** | istiod<br>tags + revisions |  waypoint (envoy) |
+
+Of these components, only waypoints (and other gateways) are intended to be operated by the AppDev (some users may choose to limit ingress gateways to the PlatEng role as well).  The remainder are the sole responsibility of the PlatEng role, and will be the focus of this reference architecture.
+
+### Tags and Revisions
+
+Istio components, particularly the waypoint, can specify which control plane they connect to (and by inference what version of the data plane they will run) using the `istio.io/rev` label set to a tag or revision.
+
+As in sidecar mode, every control plane installation may (and should) include a revision name, which is a stable identifier for that control plane installation and version.  For simplicity, we recommend using the version of the control plane as the revision name (see [./istio/control-plane-appset.yaml:9](./istio/control-plane-appset.yaml), at .spec.generators[0].list.elements[*].revision).
+
+Tags also identify control planes, but unlike revisions, tags are mutable references to revisions.  When an Istio Gateway (waypoint, ingress, or egress) references a particular tag, a dataplane is created using the version of the tag reference, and connects to the control plane indicated by the tag.  In this way, gateways can be organized into channels, or distinct groups which will be upgraded concurrently, without any involvement from the AppDev who owns the gateway.
+
+In this reference architecture, three tags are used: stable, rapid, and default (the default tag will manage any gateways which do not use the `istio.io/rev` label).  In the example application, we have included an ingress gateway on the default tag, and two waypoints for the reviews and details services, which use the rapid and stable tags.  At the time of writing, the rapid revision points to revision 1-19-3, while the stable and default revisions point to revision 1-18-5.  The tags definitions can be found at [./istio/tags.yaml](./istio/tags.yaml).
+
+## Upgrade Planning
+
+This reference architecture provides the tools to declaratively manage your Istio Ambient installations with simple pull requests.  Before performing an upgrade, however, the PlatEng team should consider how they would like their upgrades to progress.  The two most common strategies are channels and phases, and these strategies can be combined.
+
+In a phased model, there is generally a single version of Istio available in the cluster.  When a new version becomes available, the phases are moved one at a time to the new version, in order, until all phases have upgraded to the new model.  The phased model supports any number of phases based on the needs of your platform.
+
```

Comment:
With sidecars, upgrading workloads to new version could happen at user-dictated pace since it required workload re-rollout. IIUC with ambient it seems like when waypoint upgrades all proxies  behind it are upgraded at once. Is there a way to bump versions slower than shown here?

Reply:
Not today.  But I have to slightly disagree about the user-dictated pace in sidecars.  The user could set the minimum speed (ie restart all pods gradually over 24 hours), but events like deployments, scaling, and nodes going up and down can cause pods to upgrade faster - even instantaneously.  If a user would like to upgrade waypoints slowly, they should divide them into tags, and upgrade the tags one at a time over a period of time.